app/decorators/code/field.rb

```diff
@@ -78,7 +77,11 @@ def base_typescript_type
       when :boolean
         'boolean'
       when :string
-        'string'
+        if resource_attribute.enum.blank?
+          'string'
+        else
+          resource_attribute.enum.split(',').map { |x| '"' + x.strip + '"' }.join(' | ')
```

Comment:
{ |x| '"' + x.strip + '"'  }
What about `"\"#{x.strip}\""` or `'"%x"' % x.strip` ?

Reply:
I am not a ruby developer. So to be explicit, is the suggestion :

```ruby
           resource_attribute.enum.split(',').map { |x| "\"#{x.strip}\"" }.join(' | ') 
```

Reply:
Ok, done in last commit. I also corrected the existing test for typescript generation.